pkg/middleware/request_logger_test.go

```diff
@@ -1,116 +1,104 @@
-package main
+package middleware
 
 import (
 	"bytes"
 	"net/http"
 	"net/http/httptest"
-	"testing"
 
 	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/logger"
-	"github.com/stretchr/testify/assert"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
 )
 
 const RequestLoggingFormatWithoutTime = "{{.Client}} - {{.Username}} [TIMELESS] {{.Host}} {{.RequestMethod}} {{.Upstream}} {{.RequestURI}} {{.Protocol}} {{.UserAgent}} {{.StatusCode}} {{.ResponseSize}} {{.RequestDuration}}"
 
-func TestLoggingHandler_ServeHTTP(t *testing.T) {
-	tests := []struct {
+var _ = Describe("Request logger suite", func() {
+	type requestLoggerTableInput struct {
 		Format             string
 		ExpectedLogMessage string
 		Path               string
 		ExcludePaths       []string
-	}{
-		{
-			Format:             RequestLoggingFormatWithoutTime,
-			ExpectedLogMessage: "127.0.0.1 - - [TIMELESS] test-server GET - \"/foo/bar\" HTTP/1.1 \"\" 200 4 0.000\n",
-			Path:               "/foo/bar",
-			ExcludePaths:       []string{},
+	}
+
+	DescribeTable("when service a request",
+		func(in *requestLoggerTableInput) {
+			buf := bytes.NewBuffer(nil)
+			logger.SetOutput(buf)
+			logger.SetReqTemplate(in.Format)
+			logger.SetExcludePaths(in.ExcludePaths)
+
+			req, err := http.NewRequest("GET", in.Path, nil)
```

Comment:
Take a look at how we set up the request in other tests that rely on the scope, we can inject one here ish

How is the upstream being set in the scope right now? That should be done in the test handler right?